### PR TITLE
Some miscellaneous changes to support coupled global ocean sea-ice simulations

### PR DESCRIPTION
**Pull request by glwagner**
This PR contains a handful of changes made en route to coupled global ocean sea-ice simulations. Right now it just has a tweak to the interface for AbstractModel and also makes LatitudeLongitudeGrid support Flat vertical directions.

### CODE REVIEW

File: src/Models/Models.jl [hunk 1]
```diff
@@ -34,7 +34,7 @@ using Oceananigans.OutputReaders: update_field_time_series!, extract_field_times
 
 iteration(model::AbstractModel) = model.clock.iteration
 Base.time(model::AbstractModel) = model.clock.time
-architecture(model::AbstractModel) = model.architecture
+architecture(model::AbstractModel) = model.grid.architecture
```
[the remark this answers is not in the document]
Comment (glwagner) [reply src/Models/Models.jl:37]: Not all models have the `architecture` property (or rather, we don't want to require that, which this code does).

[the remark this answers is not in the document]
Comment (glwagner) [reply src/Models/Models.jl:37]: Actually we probably should not define this at all, it should be required for models to define it themselves. Although an even deeper question is why we require `architecture(model)` at all

[the remark this answers is not in the document]
Comment (navidcy) [reply src/Models/Models.jl:37]: there is also this comment somewhere...

https://github.com/CliMA/Oceananigans.jl/blob/ed5c80123506de6dbc89538c8988b06ae467b737/src/Models/Models.jl#L26-L33

perhaps `architecture(model)` was remnant from back in the day that architecture was not totally inferred from the grid?

**Comment by glwagner** [comment]
I'm closing this, the changes will be lumped into #3358

**Pull request closed without merging**